### ISSUE TITLE
wxdemo.py uses deprecated wxPython API

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#11162]
|<img alt="mcmillen's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| mcmillen reported|
|-|-|
|Trac ID|trac#1656|
|Type|defect|
|Created|2006-04-20 21:17:12Z|

"from wxPython import *" is considered deprecated by the wxPython folks:

http://wiki.wxpython.org/index.cgi/How_20to_20Learn_20wxPython

The proper thing to do is just "import wx" and change old names like "wxFoo" to "wx.Foo". 

doc/core/examples/wxdemo.py doesn't use the preferred style. So I fixed it. Attached is a patch against current SVN head (revision 16650) that changes the style of wxdemo.py appropriately.

Attachments:

* [wxdemo-style.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/7fd/7fd0dc28db1b5252f0aca04247cf21853f772e70/3f178285721a8023791e26ab07b46ab2e28a09ea.patch) (2255 bytes) - added by mcmillen on 2006-04-20 21:18:26Z - patch to upgrade deprecated style
* [wxdemo-style.2.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/7fd/7fd0dc28db1b5252f0aca04247cf21853f772e70/31c72c0fc4131d3303f8a63b9389e1035feeb538.patch) (1814 bytes) - added by exarkun on 2008-07-10 15:53:38Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__1656 1656
type__defect defect
reporter__mcmillen mcmillen
priority__low low
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1145567832000000 1145567832000000
changetime__1217186867000000 1217186867000000
version__ 
owner__ 
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="collab's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_1"></a>|collab commented|
|-|-|

Do you need an updated patch for this? Any reason this isn't merged to the trunk?

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

What version of wx will the patched wxdemo work with?  How long ago was the wxPython way deprecated?

**Comment by twisted-trac** [comment]
|<img alt="collab's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|collab removed owner|
|-|-|

Replying to [exarkun](#note_2):
> What version of wx will the patched wxdemo work with?  
[2.4 or newer](http://www.wxpython.org/migrationguide.php)

> How long ago was the wxPython way deprecated?
[24-Mar-2003](http://www.wxpython.org/CHANGES.html)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set owner to @exarkun<br>@exarkun set status to `assigned`|
|-|-|

Okay, seems like we should update, then.  The patch does need to be updated/rewritten, though, it doesn't apply at all any longer.  It should actually be much shorter now, since `wxdemo.py` does actually import from `wx` now and falls back to `wxPython.wx` on `ImportError`.  Actually, um, all this patch would do now is break support for older version of wxPython.  Well, I don't think even 2.4 works with wxreactor, so that's fine, but normally I don't think I'd accept a patch that just broke support for older versions of things.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner<br>@exarkun set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_7"></a>|@therve set owner to @exarkun|
|-|-|

Please apply.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun set status to `closed`|
|-|-|

(In [24402]) Apply wxdemo-style.2.patch

Author: exarkun
Reviewer: therve
Fixes: [#11162](https://github.com/twisted/twisted/issues/11162)

Change the wxPython integration demo to remove its attempt to import names from
`wxPython.wx`, which has been deprecated since 2003.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
